test/api/model/model_test.dart

```diff
@@ -212,4 +215,85 @@ void main() {
         .deepEquals([2, 3, 11]);
     });
   });
+
+  group('MessageEditState', () {
+    Map<String, dynamic> baseJson() => deepToJson(eg.streamMessage()) as Map<String, dynamic>;
+
+    test('edit history is absent', () {
+      check(Message.fromJson(baseJson()..['edit_history'] = null))
+        .editState.equals(MessageEditState.none);
+
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+          ..['last_edit_timestamp'] = 1678139636))
+        .editState.equals(MessageEditState.edited);
+    });
+
+    void checkEditState(MessageEditState editState, List<Map<String, dynamic>> editHistory){
+      check(Message.fromJson(baseJson()..['edit_history'] = editHistory))
+        .editState.equals(editState);
+    }
+
+    group('edit history exists', () {
+      test('Channel change only -> moved', () {
+        checkEditState(MessageEditState.moved,
+          [{'prev_stream': 5, 'stream': 7}]);
+      });
+
+      test('Topic name change only -> moved', () {
+        checkEditState(MessageEditState.moved,
+          [{'prev_topic': 'old_topic', 'topic': 'new_topic'}]);
+      });
+
+      test('Both topic and content changed -> edited', () {
+        checkEditState(MessageEditState.edited, [
+          {'prev_topic': 'old_topic', 'topic': 'new_topic'},
+          {'prev_content': 'old_content'},
+        ]);
+      });
+
+      test('Both topic and content changed in a single edit -> edited', () {
+        checkEditState(MessageEditState.edited,
+          [{'prev_topic': 'old_topic', 'topic': 'new_topic', 'prev_content': 'old_content'}]);
+      });
+
+      test('Content change only -> edited', () {
+        checkEditState(MessageEditState.edited,
+          [{'prev_content': 'old_content'}]);
+      });
+
+      test("'prev_topic' present without the 'topic' field -> moved", () {
+        checkEditState(MessageEditState.moved,
+          [{'prev_topic': 'old_topic'}]);
+      });
+    });
+
+    group('topic resolved in edit history', () {
+      test('Topic was only resolved -> none', () {
+        checkEditState(MessageEditState.none,
+          [{'prev_topic': 'old_topic', 'topic': '✔ old_topic'}]);
+      });
+
+      test('Topic was resolved but the content changed in the history -> edited', () {
+        checkEditState(MessageEditState.edited, [
+          {'prev_topic': 'old_topic', 'topic': '✔ old_topic'},
+          {'prev_content': 'old_content'},
+        ]);
+      });
+
+      test('Topic was resolved but it also changed in the history -> moved', () {
+        checkEditState(MessageEditState.moved, [
+          {'prev_topic': '✔ old_topic', 'topic': 'old_topic'},
+          {'prev_topic': 'old_topic', 'topic': 'new_topic'},
+        ]);
+      });
+
+
+      test('Unresolving topic with a weird prefix -> none', () {
+          checkEditState(MessageEditState.none,
+            [{'prev_topic': '✔ ✔old_topic', 'topic': 'old_topic'}]);
+      });
+
+    });
```

Comment:
nit: remove one empty line above this test, and one below it

Reply:
Fixed. Thanks!